src/Avalonia.Visuals/Rendering/ICustomSimpleHitTest.cs

```diff
@@ -25,7 +25,7 @@ public interface ICustomHitTest : ICustomSimpleHitTest
     public static class CustomSimpleHitTestExtensions
     {
         public static bool HitTestCustom(this IVisual visual, Point point)
-            => (visual as ICustomSimpleHitTest)?.HitTest(point) ?? visual.TransformedBounds?.Contains(point) == true;
+            => (visual as ICustomSimpleHitTest)?.HitTest(point) ?? visual.Bounds.Contains(point);
```

Comment:
Why the change to `Bounds` here? The documentation for `ICustomSimpleHitTest.HitTest` says: "The point to hit test in _global coordinate space_". `Bounds` is in the control's coordinate space, so shouldn't this stay as `TransformedBounds`?

Reply:
This change made it work, probably something else is happening, but could not find the source.

Reply:
This change doesn't change anything in your repro as far as I can tell: 

1. With deferred renderer, with and without this change it works
2. With immediate renderer, with and without this change it doesn't work

And give that this code should only apply to immediate renderer - the problem is elswhere. Pretty sure this should be `TransformedBounds`.
is elsewhere.

Reply:
Custom hit-testing doesn't seem to have been working with immediate renderer even before #5923.